### PR TITLE
vertx and spring-boot examples

### PR DESCRIPTION
**Pull request by tsmaeder**
Signed-off-by: Thomas Mäder <tmader@redhat.com>

Added devfiles from spring-boot-http-booster and vertx

### CODE REVIEW

File: dependencies/che-devfile-registry/devfiles/spring-boot-http-booster/devfile.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+projects:
+  - name: spring-boot-http-booster
+    source:
+      location: 'https://github.com/snowdrop/spring-boot-http-booster'
+      type: git
+      branch: master
+attributes:
+  persistVolumes: 'false'
+components:
+  - id: redhat/java/latest
```
Comment (tolusha) [inline dependencies/che-devfile-registry/devfiles/spring-boot-http-booster/devfile.yaml]: it references to java8
should we use `redhat/java11/latest` instead ?

File: dependencies/che-devfile-registry/devfiles/spring-boot-http-booster/devfile.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    type: chePlugin
+  - mountSources: true
+    endpoints:
+      - name: 8080/tcp
+        port: 8080
+    memoryLimit: 768Mi
+    type: dockerimage
+    volumes:
+      - name: m2
+        containerPath: /home/user/.m2
```
Comment (tolusha) [inline dependencies/che-devfile-registry/devfiles/spring-boot-http-booster/devfile.yaml]: `/home/jboss/.m2` ?

File: dependencies/che-devfile-registry/devfiles/vertx-http-booster/devfile.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    type: chePlugin
+  - mountSources: true
+    endpoints:
+      - name: 8080/tcp
+        port: 8080
+    memoryLimit: 512Mi
+    type: dockerimage
+    volumes:
+      - name: m2
+        containerPath: /home/user/.m2
```
Comment (tolusha) [inline dependencies/che-devfile-registry/devfiles/vertx-http-booster/devfile.yaml]: `/home/jboss/.m2` ?

File: dependencies/che-devfile-registry/devfiles/spring-boot-http-booster/devfile.yaml (hunk 1 above)
Comment (tsmaeder) [reply dependencies/che-devfile-registry/devfiles/spring-boot-http-booster/devfile.yaml]: no, the example is java 8, so probably explicitly referencing java 8 is correct

File: dependencies/che-devfile-registry/devfiles/spring-boot-http-booster/devfile.yaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    alias: maven
+    image: registry.redhat.io/codeready-workspaces/stacks-java-rhel8
+    env:
+      - value: >-
+          -XX:MaxRAMPercentage=50.0 -XX:+UseParallelGC -XX:MinHeapFreeRatio=10
+          -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4
+          -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true
+          -Xms20m -Djava.security.egd=file:/dev/./urandom -Duser.home=/home/user
```
[the remark this answers is not in the document]
Comment (tsmaeder) [reply dependencies/che-devfile-registry/devfiles/spring-boot-http-booster/devfile.yaml]: ???

File: dependencies/che-devfile-registry/devfiles/vertx-http-booster/devfile.yaml (hunk 3 above)
Comment (tsmaeder) [reply dependencies/che-devfile-registry/devfiles/vertx-http-booster/devfile.yaml]: yup, that is a leftover of starting with the che 7 example

File: dependencies/che-devfile-registry/devfiles/spring-boot-http-booster/devfile.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (tolusha) [reply dependencies/che-devfile-registry/devfiles/spring-boot-http-booster/devfile.yaml]: `-Duser.home=/home/user` -> `-Duser.home=/home/jboss`

File: dependencies/che-devfile-registry/devfiles/vertx-http-booster/devfile.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    alias: maven
+    image: registry.redhat.io/codeready-workspaces/stacks-java-rhel8
+    env:
+      - value: >-
+          -XX:MaxRAMPercentage=50.0 -XX:+UseParallelGC -XX:MinHeapFreeRatio=10
+          -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4
+          -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true
+          -Xms20m -Djava.security.egd=file:/dev/./urandom -Duser.home=/home/user
```
[the remark this answers is not in the document]
Comment (tolusha) [reply dependencies/che-devfile-registry/devfiles/vertx-http-booster/devfile.yaml]: -Duser.home=/home/user -> -Duser.home=/home/jboss

File: dependencies/che-devfile-registry/devfiles/spring-boot-http-booster/devfile.yaml (hunk 1 above)
Comment (nickboldt) [reply dependencies/che-devfile-registry/devfiles/spring-boot-http-booster/devfile.yaml]: then please use `java8` not `java` as some day java might mean `java11`.

**Pull request merged**